scripts/build/esbuild-plugins/primitive-define.mjs

```diff
@@ -0,0 +1,32 @@
+function stringify(value) {
+  if (
+    value !== null &&
+    !["boolean", "number", "string", "undefined"].includes(typeof value)
+  ) {
+    throw Object.assign(new TypeError("value not allowed"), { value });
+  }
+
+  return JSON.stringify(value);
+}
+
+/*
+Non-primitive value will cause esbuild inject a function call in every file
```

Comment:
I'm not familiar with esbuild, so I have a question. Should we allow arrays and objects that includes only primitives.

Reply:
No, only primitives, that's extractly the use case I'm removing https://github.com/prettier/prettier/pull/13728/files#diff-af8da34bed4dccd1b684852e1795a05ec40e317354315bfa900d10f48e760493L112